integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/MergeHandler.scala

```diff
@@ -0,0 +1,213 @@
+    val updatedRows = stats.updatedRows
+    val uuid = UUID.randomUUID.toString
+    job.setJobID(new JobID(uuid, 0))
+    val path = targetCarbonTable.getTablePath + CarbonCommonConstants.FILE_SEPARATOR + "avro"
+    FileOutputFormat.setOutputPath(job, new Path(path))
+    val factory = AvroFileFormatFactory.getAvroWriter(sparkSession, job, schema)
+    val config = SparkSQLUtil.broadCastHadoopConf(sparkSession.sparkContext, job.getConfiguration)
+    frame.queryExecution.toRdd.mapPartitionsWithIndex { case (index, iterator) =>
+      val confB = config.value.value
+      val task = new TaskID(new JobID(uuid, 0), TaskType.MAP, index)
+      val attemptID = new TaskAttemptID(task, index)
+      val context = new TaskAttemptContextImpl(confB, attemptID)
+      val writer = factory.newInstance(path + CarbonCommonConstants.FILE_SEPARATOR + task.toString,
+        schema, context)
+      new Iterator[InternalRow] {
+        override def hasNext: Boolean = {
+          if (iterator.hasNext) {
+            true
+          } else {
+            writer.close()
+            false
+          }
+        }
+
+        override def next(): InternalRow = {
+          val row = iterator.next()
+          val newArray = new Array[Any](1)
+          val tupleID = row.getUTF8String(tupleId)
+          if (tupleID == null) {
+            insertedRows.add(1)
+          } else {
+            newArray(0) = tupleID
+            writer.write(new GenericInternalRow(newArray))
+            updatedRows.add(1)
+          }
+          null
+        }
+      }
+    }.count()
+    val deltaRdd = AvroFileFormatFactory.readAvro(sparkSession, path)
+    (deltaRdd, path)
+  }
+
+  protected def triggerAction(factTimestamp: Long,
+      executorErrors: ExecutionErrors,
+      deltaRdd: RDD[Row],
+      deltaPath: String): (util.List[SegmentUpdateDetails], Seq[Segment]) = {
+    val tuple = MergeUtil.triggerAction(sparkSession,
+      targetCarbonTable,
+      factTimestamp,
+      executorErrors,
+      deltaRdd)
+    FileFactory.deleteAllCarbonFilesOfDir(FileFactory.getCarbonFile(deltaPath))
+    MergeUtil.updateSegmentStatusAfterUpdateOrDelete(targetCarbonTable, factTimestamp, tuple)
+    tuple
+  }
+
+  protected def insertDataToTargetTable(updateTableModel: Option[UpdateTableModel]): Seq[Row] = {
+    val tableCols =
+      targetCarbonTable.getCreateOrderColumn.asScala.map(_.getColName).
+        filterNot(_.equalsIgnoreCase(CarbonCommonConstants.DEFAULT_INVISIBLE_DUMMY_MEASURE))
+    val header = tableCols.mkString(",")
+    val dataFrame = srcDS.select(tableCols.map(col): _*)
+    MergeUtil.insertDataToTargetTable(sparkSession,
+      targetCarbonTable,
+      header,
+      updateTableModel,
+      dataFrame)
+  }
+
+  protected def tryHorizontalCompaction(): Unit = {
+    // Do IUD Compaction.
+    HorizontalCompaction.tryHorizontalCompaction(
+      sparkSession, targetCarbonTable)
+  }
+
+  def handleMerge()
+}
+
+case class UpdateHandler(sparkSession: SparkSession,
```

Comment:
move param to next line, please follow the coding convention

Reply:
done